### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by markuskramerIgitt**
Improvements:
- Omits the install dir dialog
  - See [WixUI_Mondo-HelloWorldDialog](https://github.com/markuskramerIgitt/LearnWix3/blob/master/WixUI_Mondo-HelloWorldDialog/Product.wxs) for details
- Refactored lifetime data removal: WiX XML replaces C#
  - lifetime data means: data not from the installer
  - WiX XML operates transaction save and more robust than C#
  - Closes issue https://github.com/markuskramerIgitt/salt-windows-msi/issues/77
- VC++ Runtime 2013 added (for OpenSSL) 
- VC++ Runtime 2008 removed (potentially for Python2, but no use case known)
- REMOVE_CONFIG replaces KEEP_CONFIG
- Pauses Virus Real-time protection during (un)install
- Utilizes setupWixEnvironment.ps1
- Uses Wix 3.11.2
- Documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,18 +41,13 @@ Salt Minion-specific msi-properties:
...
  `INSTALLFOLDER`        | `C:\salt\`              | Where to install the Minion  __DO NOT CHANGE (yet)__  --- __BLOCKED BY__ [issue#38430](https://github.com/saltstack/salt/issues/38430)
 
 
-These files and directories are regarded as config and kept:
-
-- C:\salt\conf\minion
-- C:\salt\conf\minion.d\
-- c:\salt\var\cache\salt\minion\extmods\
-- c:\salt\var\cache\salt\minion\files\
+The config directory is C:\salt\conf\minion
```
Comment (twangboy) [inline README.md:50]: The config directory is `C:\salt\conf`

File: documentation/idea_KEEP_CONFIG.md [hunk 2]
```diff
@@ -11,13 +11,12 @@ The registry seems to be write protected for msi custom actions at uninstall tim
...
 
-On uninstall, we store a plain text file
+On uninstall, we could store a plain text file
 `%SystemDrive%\ProgramData\SaltStack\SaltMinion\KEPT_CONFIG`
 which contains the path to the (former) INSTALLDIR/salt/conf.
 
-This file is only created on uninstall with `KEEP_CONFIG=1` (the default).
-On uninstall with KEEP_CONFIG=0, no configuration remains. 
-Maybe the logs should remain.
+This file could be be created on default uninstall without `REMOVE_CONFIG` (the default).
```
Comment (twangboy) [inline documentation/idea_KEEP_CONFIG.md:18]: be be

File: README.md (hunk 1 above)
Comment (twangboy) [reply README.md:50]: Not sure exactly what you're trying to say here

Comment (markuskramerIgitt) [reply README.md]: I removed the sentence in https://github.com/saltstack/salt-windows-msi/pull/48/commits/eec012888e6c104314d68c8db9a1f6682be4a027

File: documentation/idea_KEEP_CONFIG.md (hunk 2 above)
Comment (markuskramerIgitt) [reply documentation/idea_KEEP_CONFIG.md]: Thank you, I removed one "be" in https://github.com/saltstack/salt-windows-msi/pull/48/commits/eec012888e6c104314d68c8db9a1f6682be4a027

File: wix.d/MinionMSI/Product.wxs [hunk 3]
```diff
@@ -149,30 +141,19 @@ xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
...
 
     <Feature Id="ProductFeature" Title="Minion" Level="1">
       <ComponentGroupRef Id="ProductComponents" />
-      <ComponentRef      Id="CleanupBinFolder" />
-      <Feature Id="VC90_CRT_x86_x64" Title="Visual C++ 9 64-bit Runtime" AllowAdvertise="no" Display="hidden" Level="1">
-        <MergeRef Id="MSM_VC90_CRT_64"/>
+      <ComponentRef      Id="RemoveFolderEx_BINFOLDER_Component" />
+      <ComponentRef      Id="RemoveFolderEx_VARFOLDER_Component"/>
+
+      <Feature Id="VC90_CRT_x86_x64" Title="Visual C++ 2013 64-bit Runtime" AllowAdvertise="no" Display="hidden" Level="1">
```
Comment (twangboy) [inline wix.d/MinionMSI/Product.wxs:155]: Should probably be `VC120_CRT_x64`

File: build_env.ps1 [hunk 4]
```diff
@@ -39,71 +18,27 @@ $ignore_bool = (Test-Path -Path $salt_msi_resources) -Or (New-Item -ItemType dir
...
+## VC++ Runtime merge modules from Visual Studio 2013 SP5.
+VerifyOrDownload "c:/salt_msi_resources/Microsoft_VC120_CRT_x64.msm" `
+    "http://repo.saltstack.com/windows/dependencies/64/Microsoft_VC120_CRT_x64.msm" `
+    "15FD10A495287505184B8913DF8D6A9CA461F44F78BC74115A0C14A5EDD1C9A7"
+
+VerifyOrDownload "c:/salt_msi_resources/Microsoft_VC120_CRT_x86.msm" `
+    "http://repo.saltstack.com/windows/dependencies/32/Microsoft_VC120_CRT_x86.msm" `
+    "26340B393F52888B908AC3E67B935A80D390E1728A31FF38EBCEC01117EB2579"
+
+## VC++ Runtime merge module from Visual Studio 2008 SP2.
```
Comment (twangboy) [inline build_env.ps1:32]: Can probably remove this since it's not longer being installed.

File: wix.d/MinionMSI/Product.wxs [hunk 5]
```diff
@@ -149,30 +141,19 @@ xmlns:util="http://schemas.microsoft.com/wix/UtilExtension">
...
-        <RegistryValue Root="HKLM" Key="SOFTWARE\Saltstack\Salt Minion" Name="BinPath" Type="string" Value="[BINFOLDER]" KeyPath="yes" /> <util:RemoveFolderEx On="uninstall" Property="BINFOLDER" />
-      </Component>
-    </DirectoryRef>
-
-
     <DirectoryRef Id="TARGETDIR">
-      <Merge Id="MSM_VC90_CRT_64" SourceFile="c:\salt_msi_resources\Microsoft_VC90_CRT_x86_x64.msm" DiskId="1" Language="0"/>
+    <?if $(var.TargetPlatform)=amd64 ?>
+      <Merge Id="MSM_VC120_CRT" SourceFile="c:\salt_msi_resources\Microsoft_VC120_CRT_x64.msm" DiskId="1" Language="0"/>
+    <?endif ?>
```
Comment (twangboy) [inline wix.d/MinionMSI/Product.wxs:147]: Do we need to add something here to install the 32bit merge module?

File: wix.d/MinionMSI/Product.wxs (hunk 3 above)
Comment (markuskramerIgitt) [reply wix.d/MinionMSI/Product.wxs:155]: Already changed to 120.

File: wix.d/MinionMSI/Product.wxs (hunk 5 above)
Comment (markuskramerIgitt) [reply wix.d/MinionMSI/Product.wxs:147]: Yes, namely: 

 ```
    <?else ?>
   <Merge Id="MSM_VC120_CRT" SourceFile="c:\salt_msi_resources\Microsoft_VC120_CRT_x86.msm" DiskId="1" Language="0"/>
```

But 32bit is out of my scope

File: build_env.ps1 (hunk 4 above)
Comment (markuskramerIgitt) [reply build_env.ps1:32]: This is 32bit not for Python 2 but for OpenSSL. 
You can remove, if you wish, I have no interest in 32-bit.

**Pull request merged**